src/OVAL/probes/unix/sysctl.c

```diff
@@ -149,9 +165,21 @@ int probe_main(probe_ctx *ctx, void *probe_arg)
                         l = fread(sysval, 1, sizeof sysval - 1, fp);
 
                         if (ferror(fp)) {
-                                dE("An error ocured when reading from \"%s\" (fp=%p): l=%ld, %u, %s\n",
-                                    mibpath, fp, l, errno, strerror(errno));
-                                goto fail_item;
+				/* Linux 4.1.0 introduced a per-NIC IPv6 stable_secret file.
+				 * The stable_secret file cannot be read until it is set,
+				 * so we skip it.
+				 */
+				if (strncmp(ofts_ent->path, ipv6_conf_path, ipv6_conf_path_len) == 0 &&
```

Comment:
@mpreisler This was my original idea. Originally I have checked for this path before fopen(), I have had this block of code somewhere about line 118. I have already discussed this with @isimluk. We realized that it would skip the file also if it is possible to read from it. When someone has written an ipv6 address to this file, it is possible to read it, and I think in such case the probe should collect it. For this use-case, I decided to move this check here.